aviatrix/resource_aviatrix_aws_tgw_vpc_attachment.go

```diff
@@ -245,6 +266,38 @@ func resourceAviatrixAwsTgwVpcAttachmentUpdate(d *schema.ResourceData, meta inte
 		}
 	}
 
+	if d.HasChange("edge_attachment") {
+		awsTgwVpcAttachment := &goaviatrix.AwsTgwVpcAttachment{
+			TgwName:            d.Get("tgw_name").(string),
+			VpcID:              d.Get("vpc_id").(string),
+			SecurityDomainName: d.Get("security_domain_name").(string),
+			EdgeAttachment:     d.Get("edge_attachment").(string),
+		}
+
+		oldEA, newEA := d.GetChange("edge_attachment")
+
+		if oldEA != "" && newEA != "" {
+			awsTgwVpcAttachment.EdgeAttachment = ""
+
+			err := client.UpdateFirewallAttachmentAccessFromOnprem(awsTgwVpcAttachment)
+			if err != nil {
+				return fmt.Errorf("failed to disable firewall attachment access from onprem while updating: %s", err)
+			}
+
+			awsTgwVpcAttachment.EdgeAttachment = d.Get("edge_attachment").(string)
+
+			err = client.UpdateFirewallAttachmentAccessFromOnprem(awsTgwVpcAttachment)
+			if err != nil {
+				return fmt.Errorf("failed to enable firewall attachment access from onprem while updating: %s", err)
+			}
+		} else {
```

Comment:
What about the case where we are going from "" -> "someID", do we need to validate again that this is a FireNet attachment?

Or would the API return an error? If that's the case we probably don't need to validate again

Reply:
API will return an error, but we can add a validation here.